MozillaFoundation/foundation.mozilla.org
# Moved localized logic from take_action.html template to base.py with …

# Description
This PR removes `.localized` from `take_action.html` and moved the logic to `base.py`, implementing `get_localized_take_action_cards` function, that iterates over `take_action_cards` and add every card to a list, as well as the now localized `internal_link`. This PR also updates the `take_action.html` template, adding the new context variable `localized_take_action_cards`.

Link to sample test page: https://foundation-s-tp1-103-re-yv4aeu.herokuapp.com/en/
Related PRs/issues: https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10010

# How to test
- Scroll down Homepage until Take action section.
- Change the code locale in the URL and make sure the external link properly keeps the updated locale.

Example:
![image](https://github.com/user-attachments/assets/a9984ca2-770e-404e-a0e6-f079a94f93ca)

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-1364)
